src/ScriptEditor/ui/ThemeEditorModal.tsx

```diff
@@ -74,12 +75,19 @@ export function ThemeEditorModal(props: ThemeEditorProps): React.ReactElement {
 
   function onThemeChange(event: React.ChangeEvent<HTMLInputElement>): void {
     try {
-      const importedTheme = JSON.parse(event.target.value);
-      if (typeof importedTheme !== "object") return;
+      const importedTheme = JSON.parse(event.target.value) as typeof Settings.EditorTheme;
+      if (importedTheme == null) {
+        throw new Error("Theme data must not be null or undefined.");
+      }
+      if (typeof importedTheme !== "object") {
+        throw new Error(`Theme data is invalid.`);
+      }
       Settings.EditorTheme = importedTheme;
```

Comment:
Technically you should do the `as` cast here, since it's a bit gross to assert it's an EditorTheme and then be checking if it's null after.

Reply:
Noted. I plan to make a PR that validates the input theme data instead of bypassing the type check by casting the result of `JSON.parse`. Doing that may be a bit complicated (the JSON schema of the editor theme is more complicated than the game theme's JSON), so I have not done it.

Reply:
Yeah, properly validating JSON schemas is tricky and annoying.